functions/src/list_index.ts

```diff
@@ -5,195 +5,247 @@ import { db } from ".";
+        );
+        if (deleteRequest !== "true") {
+          let counter = 0;
+          let indexMap = new Map<DocumentReference, boolean>();
+          let batch = db.batch();
+          const reference = await db.collection("list").doc(listId);
+          const indexConfigs = await reference.collection("indexConfigs").get();
+          const items = await reference.collection("item").get();
+          await statusRef.update({ total: items.size });
+          for (let indexConfig of indexConfigs.docs) {
+            let entityIndexFields = await indexConfig.ref
+              .collection("entityIndexFields")
+              .orderBy("createdTimestamp")
+              .get();
+            // var valid = true;
+            // for (let entityIndexField of entityIndexFields.docs) {
+            //   if (!entityIndexField.data().valid) {
+            //     valid = false;
+            //     break;
+            //   }
+            // }
+            if (
+              //valid &&
+              entityIndexFields.docs.length > 0
+            ) {
+              let type = indexConfig.data().type;
+              for (let item of items.docs) {
+                if (type === "Single field") {
+                  let value =
+                    item.data()[entityIndexFields.docs[0].data().value];
+                  if (!Array.isArray(value)) {
+                    let indexDocRef = null;
+                    if (value != "undefined") {
+                      indexDocRef = addToBatch(
+                        batch,
+                        listId,
+                        type,
+                        item.ref,
+                        value
+                      );
+                    }
+                    counter++;
+                    if (indexDocRef !== undefined) indexMap.set(item.ref, true);
+                    if (counter > 490) {
+                      statusRef.update({ count: [...indexMap.keys()].length });
+                      await batch.commit();
+                      batch = db.batch();
+                      counter = 0;
+                    }
                   }
-                  name += (name.length > 0 ? ' ' : '') + value;
-                }
-                if (!containsArray) {
-                  const indexDocRef= addToBatch(batch, listId, type, item.ref, name);
-                  counter++;
-                  if(indexDocRef!==undefined)
-                    indexMap.set(item.ref, true);
-                  if (counter > 490) {
-                    statusRef.update({count: [...indexMap.keys()].length});
-                    await batch.commit();
-                    batch = db.batch();
-                    counter = 0;
+                } else if (type === "Multiple fields") {
+                  var containsArray = false;
+                  var name = "";
+                  for (let entityIndexField of entityIndexFields.docs) {
+                    let value = item.data()[entityIndexField.data().value];
```

Comment:
I think the problem is actually rooted here where there is no value in the document, so we are getting undefined, which then gets converted to string 'undefined' when we append it to the string.

Reply:
Refactored code to resolve this issue